### PR TITLE
[4.0] Handle application class deprecations

### PR DESCRIPTION
**Pull request by mbabker**
### Summary of Changes

Handle deprecations in the application library
### Testing Instructions

CMS functions still
### Documentation Changes Required

Document deprecations

### CODE REVIEW

File: libraries/cms/application/cms.php [hunk 1]
```diff
@@ -52,28 +52,25 @@ class JApplicationCms extends JApplicationWeb implements ContainerAwareInterface
 	 * The client identifier.
 	 *
 	 * @var    integer
-	 * @since  3.2
-	 * @deprecated  4.0  Will be renamed $clientId
+	 * @since  4.0
```
Comment (wilsonge) [inline libraries/cms/application/cms.php]: Please use `__DEPLOY_VERSION` :)

Comment (zero-24) [inline libraries/cms/application/cms.php]: `__DEPLOY_VERSION__` :smile:

**Comment by zero-24** [comment]
@mbabker why did you close this? and where is @wilsonge 's comment?

**Comment by mbabker** [comment]
Brian, George, and I are having an in-person debate over how to handle the deprecation of getCfg and getPageParameters.

**Comment by wilsonge** [comment]
Merged with https://github.com/joomla/joomla-cms/commit/cf2ed115fa86d058719ebe7c39767bec69dee937

**Comment by zero-24** [comment]
Ah ok I thought you are on a wedding and not a codesprint ;). @wilsonge so we add this PR to the milestone or not? I'm confused.

**Comment by mbabker** [comment]
The wedding's tomorrow.  Today's apparently all business day.

**Comment by wilsonge** [comment]
I'm wearing swimshorts by the pool AND merging PR's. It's efficiency :D

**Comment by zero-24** [comment]
have fun in Greece :smile:

**Comment by mbabker** [comment]
And they are a scary orange color.  @MATsxm would approve.

**Comment by wilsonge** [comment]
🏊

**Comment by MATsxm** [comment]
Do I really have to express myself on GH or should I be polite as fuck :smirk:

**Pull request closed without merging**